corehq/apps/domain/forms.py

```diff
@@ -1805,8 +1805,20 @@ def process_subscription_management(self):
             self.domain, edition=self.cleaned_data['software_plan_edition'],
         )
 
-        if not self.current_subscription or self.cleaned_data['start_date'] > datetime.date.today():
-            new_subscription = Subscription.new_domain_subscription(
+        if (
+            self.current_subscription
+            and self.current_subscription.service_type == SubscriptionType.CONTRACTED
+            and self.current_subscription.plan_version == new_plan_version
+            and self.current_subscription.date_start == self.cleaned_data['start_date']
```

Comment:
nit: this would be much easier to read if this was named
```py
if (contracted_subscription_already_exists): 
    do_thing()
else:
   do_other_thing()
```